gateway/src/main/java/com/hurence/webapiservice/http/grafana/GrafanaApiImpl.java

```diff
@@ -116,6 +127,78 @@ at initialization (did not successfully reproduced this in a unit test).//TODO
                 }).subscribe();
     }
 
+    @Override
+    public void export(RoutingContext context) {
+        final long startRequest = System.currentTimeMillis();
+        final QueryRequestParam request;
+        try {
+            JsonObject requestBody = context.getBodyAsJson();
+            /*
+                When declaring QueryRequestParser as a static variable, There is a problem parsing parallel requests
+                at initialization (did not successfully reproduced this in a unit test).//TODO
+             */
+            request = new QueryRequestParser().parseRequest(requestBody);
+        } catch (Exception ex) {
+            LOGGER.error("error parsing request", ex);
+            context.response().setStatusCode(BAD_REQUEST);
+            context.response().setStatusMessage(ex.getMessage());
+            context.response().putHeader("Content-Type", "application/json");
+            context.response().end();
+            return;
+        }
+
+        final JsonObject getTimeSeriesChunkParams = buildHistorianRequest(request);
+
+        service
+                .rxGetTimeSeries(getTimeSeriesChunkParams)
+                .map(sampledTimeSeries -> {
+                    JsonArray timeseries = sampledTimeSeries.getJsonArray(TIMESERIES_RESPONSE_FIELD);
+                    if (LOGGER.isDebugEnabled()) {
+                        timeseries.forEach(metric -> {
+                            JsonObject el = (JsonObject) metric;
+                            String metricName = el.getString(TimeSeriesExtracterImpl.TIMESERIE_NAME);
+                            int size = el.getJsonArray(TimeSeriesExtracterImpl.TIMESERIE_POINT).size();
+                            LOGGER.debug("[REQUEST ID {}] return {} points for metric {}.",
+                                    request.getRequestId(),size, metricName);
+                        });
+                        LOGGER.debug("[REQUEST ID {}] Sampled a total of {} points in {} ms.",
+                                request.getRequestId(),
+                                sampledTimeSeries.getLong(TOTAL_POINTS_RESPONSE_FIELD, 0L),
+                                System.currentTimeMillis() - startRequest);
+                    }
+                    return timeseries;
+                })
+                .map(timeseries -> {
+                    QueryResponse queryResponse = new QueryResponse(timeseries);
```

Comment:
This naming is not explicit at all in my Opinion. Why QueryResponse ?
What is the purpose of your object ? You should name it depending on what it Does.

Reply:
done, name has changed